### PR TITLE
Add phone input

### LINKED ISSUE #371: Add phone input to checkout
- [x] Add phone number input to the Checkout flow.
- [x] Validate phone number.

In checkout designs there is a [nice phone input](https://www.figma.com/file/oBQhZFwyFcXauxFrb3mbRb/saleor_checkout?node-id=2482%3A11686). Let's implement it in ui-kit so it's more user friendly.

### PR DESCRIPTION
**Pull request by bmigirl**
Fixes #371

### CODE REVIEW

File: packages/checkout-storefront/package.json [hunk 1]
```diff
@@ -44,7 +44,9 @@
     "checkout-common": "workspace:*",
     "clsx": "1.2.1",
     "concurrently": "7.5.0",
+    "fs": "0.0.1-security",
```
Comment (mmiszy) [inline packages/checkout-storefront/package.json:47]: What does it do?

File: packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        });
+
+        result = {
+          ...result,
+          values,
+        };
+      } catch (error) {
+        const errors = getErrorsAsObject(getAllValidationErrors(error as YupValidationErrorObject));
+        result = { ...result, errors };
+      }
```
Comment (mmiszy) [inline packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts:46]: I think we could extract it to a function `getAddressFormResolverValue` or something, and avoid using `let` then.

File: packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+      const phoneRequiredError = result.errors?.phone?.type === "required";
+
+      if (!isValidNumber(data.phone, data.countryCode) && !phoneRequiredError) {
+        result = {
```
Comment (mmiszy) [inline packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts:51]:
```suggestion
        return {
```

File: packages/checkout-storefront/src/hooks/__tests__/usePhoneNumberUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  it("should return false for invalid phone numbers", () => {
+    const { result: hook } = renderHook(() => usePhoneNumberUtils());
+
+    expect(hook.current.isValidNumber("+4172268096", countryCode)).toEqual(false);
+    expect(hook.current.isValidNumber("+48111111111", countryCode)).toEqual(false);
+    expect(hook.current.isValidNumber("123", countryCode)).toEqual(false);
+    expect(hook.current.isValidNumber("foobar", countryCode)).toEqual(false);
+    expect(hook.current.isValidNumber("+48", countryCode)).toEqual(false);
+  });
+});
```
Comment (mmiszy) [inline packages/checkout-storefront/src/hooks/__tests__/usePhoneNumberUtils.test.ts:23]: Let's also add tests for `getPhoneNumberWithCountryCode`

File: packages/checkout-storefront/src/hooks/usePhoneNumberUtils.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return `+${callingCode}${phone}`;
+      }
+
+      return phone;
+    },
+    [getPhoneNumberInstance, isMissingCountryCallingCode]
+  );
+
+  return { isValidNumber, getPhoneNumberWithCountryCode };
+};
```
Comment (mmiszy) [inline packages/checkout-storefront/src/hooks/usePhoneNumberUtils.ts:57]: I think this doesn't even need to be a React hook. Just plain old utility functions. What do you think? No need for `useCallback`s and easier to test.

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts (hunk 2 above)
Comment (bmigirl) [reply packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts:46]: tbh I added it like that knowing that it's like the resolvers will be refactored soon along with changing the forms lib so for now I'd just leave it as it is

File: packages/checkout-storefront/src/hooks/usePhoneNumberUtils.ts (hunk 5 above)
Comment (bmigirl) [reply packages/checkout-storefront/src/hooks/usePhoneNumberUtils.ts:57]: Oh yeah, you're right. I added it like this when testing with another phone number lib and it required to be hook, later forgot to change it.

**Review by mmiszy — DISMISSED** [review]

**Comment by mmiszy** [comment]
You forgot to run `pnpm i` after removing that one dependency, other than that looks good!

File: packages/checkout-storefront/src/lib/utils/phoneNumber.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  } catch (error) {
+    return null;
+  }
+};
+
+export const isValidPhoneNumber = (phone: string, countryCode: CountryCode | undefined) =>
+  !!getPhoneNumberInstance(phone, countryCode)?.isValid();
+
+const isMissingCountryCallingCode = (phone: string, countryCode: CountryCode) => {
+  const isValidWithoutCountryCode = getPhoneNumberInstance(phone, undefined);
```
Comment (witoszekdev) [inline packages/checkout-storefront/src/lib/utils/phoneNumber.ts:25]: Why do we use `getPhoneNumberInstance` instead of `isValidPhoneNumber` here?

**Comment by michalina-graczyk** [comment]
When I go to checkout there is 404. Please check https://watch.screencastify.com/v/bZottsmZZMUV2EWhCAA0

**Review by mmiszy — APPROVED** [review]

**Pull request merged**